src/test/kotlin/no/nav/syfo/behandlerdialog/api/BehandlerdialogApiSpek.kt

```diff
@@ -0,0 +1,144 @@
+package no.nav.syfo.behandlerdialog.api
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import io.ktor.http.*
+import io.ktor.server.testing.*
+import no.nav.syfo.testhelper.*
+import no.nav.syfo.testhelper.generator.generateBehandlerdialog
+import no.nav.syfo.util.*
+import org.amshove.kluent.shouldBeEqualTo
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+class BehandlerdialogApiSpek : Spek({
+    val objectMapper: ObjectMapper = configuredJacksonMapper()
+
+    describe(BehandlerdialogApiSpek::class.java.simpleName) {
+        with(TestApplicationEngine()) {
+            start()
+            val externalMockEnvironment = ExternalMockEnvironment.instance
+            application.testApiModule(
+                externalMockEnvironment = externalMockEnvironment,
+            )
+
+            val urlGet = behandlerdialogApiBasePath
+            val urlPost = "$behandlerdialogApiBasePath$behandlerdialogApiPersonidentPath"
+
+            val validToken = generateJWT(
+                audience = externalMockEnvironment.environment.azure.appClientId,
+                issuer = externalMockEnvironment.wellKnownInternalAzureAD.issuer,
+                navIdent = UserConstants.VEILEDER_IDENT,
+            )
+
+            describe("Get behandlerdialoger for person") {
+                describe("Happy path") {
+                    it("Returns OK from GET") {
+                        with(
+                            handleRequest(HttpMethod.Get, urlGet) {
+                                addHeader(HttpHeaders.Authorization, bearerHeader(validToken))
+                                addHeader(NAV_PERSONIDENT_HEADER, UserConstants.ARBEIDSTAKER_PERSONIDENT.value)
+                            }
+                        ) {
+                            response.status() shouldBeEqualTo HttpStatusCode.OK
+                        }
+                    }
+                }
+
+                describe("Unhappy path") {
+                    it("Returns status Unauthorized if no token is supplied") {
+                        testMissingToken(urlGet, HttpMethod.Get)
+                    }
+                    it("returns status Forbidden if denied access to person") {
+                        testDeniedPersonAccess(urlGet, validToken, HttpMethod.Get)
+                    }
+                    it("returns status BadRequest if no $NAV_PERSONIDENT_HEADER is supplied") {
+                        testMissingPersonIdent(urlGet, validToken, HttpMethod.Get)
+                    }
+                    it("returns status BadRequest if $NAV_PERSONIDENT_HEADER with invalid PersonIdent is supplied") {
+                        testInvalidPersonIdent(urlGet, validToken, HttpMethod.Get)
+                    }
+                }
+            }
+
+            describe("Create new behandlerdialog") {
```

Comment:
Burde vel få med melding i testnavnene her (new melding in behandlerdialog e.l.)